static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,448 @@
+  const seen = new Set();
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "interactingProtein") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let protein_name = null;
+      let confidence_score = null;
+      let parent_protein = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "name") {
+          // check for null name values and move to next iteration if true
+          if (_.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          protein_name = n.nodes[0].value;
+          parent_protein = nodeName;
+        } else if (n.property === "confidenceScore") {
+          // not checking for empty values because if name exists, confidence score must exist
+          for (const n1 of n.nodes) {
+            for (const n2 of n1.neighbors) {
+              if (n2.property === "value") {
+                const num = Number(n2.nodes[0].value);
+                if (num <= 1) {
+                  confidence_score = num;
+                }
+              }
+            }
+          }
+        }
+      }
+      // checking for duplicates
+      if (!seen.has(protein_name)) {
+        returnData.push({
+          name: protein_name,
+          value: confidence_score,
+          parent: parent_protein,
+        });
+        seen.add(protein_name);
+      }
+    }
+    return returnData;
+  }
+  return [];
+}
+
+export function getDiseaseGeneAssoc(data: GraphNodes): ProteinNumData[] {
+  // Disease Gene Associations
+  if (!data) {
+    return [];
+  }
+  const returnData: ProteinNumData[] = [];
+  const seen = new Set();
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "geneID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let score = null;
+      let disease = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property !== "geneID") {
+          continue;
+        }
+        for (const n1 of n.nodes) {
+          for (const n2 of n1.neighbors) {
+            if (n2.property === "diseaseOntologyID") {
+              for (const n3 of n2.nodes) {
+                if (n3.neighbors === undefined) {
+                  continue;
+                }
+                for (const n4 of n3.neighbors) {
+                  if (n4.property !== "commonName") {
+                    continue;
+                  }
+                  if (_.isEmpty(n4.nodes[0].value)) {
```

Comment:
all cases where you are reading the value of a specific node in the nodes list, check that the nodes list contains that node first.